plugins/zoom/icon_zoom/tasks/monitor_sign_in_out_activity/task.py

```diff
@@ -45,20 +54,25 @@ def __init__(self):
 
     # pylint: disable=unused-argument
     def run(self, params={}, state={}):
-        # Check if first run
-        if not state.get(self.LAST_REQUEST_TIMESTAMP):
-            self.logger.info("First run")
-            output, new_state = self.first_run(state=state)
-        else:
-            # Subsequent run
-            self.logger.info("Subsequent run")
-            output, new_state = self.subsequent_run(state=state)
-
-        # Turn events list back into a list of dicts
-        output = [event.__dict__ for event in output]
-        return output, new_state
-
-    def first_run(self, state: dict) -> ([dict], dict):
+        try:
+            # Check if first run
+            if not state.get(self.LAST_REQUEST_TIMESTAMP):
+                self.logger.info("First run")
+                output, new_state, has_more_pages, status_code, error = self.first_run(state=state)
+            else:
+                # Subsequent run
+                self.logger.info("Subsequent run")
+                output, new_state, has_more_pages, status_code, error = self.subsequent_run(state=state)
+
+            # Turn events list back into a list of dicts
+            output = [event.__dict__ for event in output]
+            return output, new_state, has_more_pages, status_code, error
+
+        except Exception as error:
+            self.logger.error(f"Unhandled exception occurred during {self.name} task: {error}")
+            return PluginException(preset=PluginException.Preset.UNKNOWN, data=error)
```

Comment:
```suggestion
            return [], {}, False, 500, PluginException(preset=PluginException.Preset.UNKNOWN, data=error)
```
I think we'll need to return like this, but I'm unsure of what we want to do with the state here.